component/telemetry.go

```diff
@@ -29,4 +29,20 @@ type TelemetrySettings struct {
 
 	// Resource contains the resource attributes for the collector's telemetry.
 	Resource pcommon.Resource
+
+	// ReportComponentStatus allows a component to report runtime changes in status. The service
+	// will automatically report status for a component during startup and shutdown. Components can
+	// use this method to report status after start and before shutdown. ReportComponentStatus
+	// will only return errors if the API used incorrectly. The three scenarios where an error will
```

Comment:
```suggestion
	// will only return errors if the API used incorrectly. The two scenarios where an error will
```
There are only two scenarios listed, is there one missing or should this be updated?

Reply:
There were three scenarios, but after some refactoring, there are only two. Thanks for correcting.